src/libs/github/file_info.ts

```diff
@@ -64,21 +66,36 @@ export const resolveDiffFileInfo = (codeView: HTMLElement): Observable<FileInfo>
         }))
     )
 
-export const resolveFileInfo = (codeView: HTMLElement): Observable<FileInfo> =>
-    of(codeView).pipe(
-        map(() => {
-            const { repoPath, filePath, rev } = parseURL()
-
-            return { repoPath, filePath, rev }
-        }),
-        filter(propertyIsDefined('filePath')),
-        switchMap(({ repoPath, rev, ...rest }) =>
-            resolveRev({ repoPath, rev }).pipe(
-                retryWhenCloneInProgressError(),
-                map(commitID => ({ ...rest, repoPath, commitID, rev: rev || commitID }))
+export const resolveFileInfo = (): Observable<FileInfo> => {
+    const { repoPath, filePath } = parseURL()
+    if (!filePath) {
+        return throwError(
+            new Error(
+                `Unable to determine the file path of the current file because the current URL (window.location ${
+                    window.location
+                }) does not have a file path.`
             )
         )
-    )
+    }
+
+    const commitID = commitIDFromPermalink({
+        selector: '.js-permalink-shortcut',
+        hrefRegex: new RegExp('^/.*?/.*?/blob/([0-9a-f]{40})/'),
+    })
+
+    if (isErrorLike(commitID)) {
```

Comment:
Throwing the error inside `commitIDFromPermalink` would prevent the need for this, but you could add a check for whether its not a string here to prevent the need for the separate declaration below

```suggestion
    if (isErrorLike(commitID) || typeof commitID !== 'string') {
```